sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -290,8 +284,8 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
         df,
         dataSource.createSink(outputMode),
         outputMode,
-        useTempCheckpointLocation = useTempCheckpointLocation,
-        recoverFromCheckpointLocation = recoverFromCheckpointLocation,
+        useTempCheckpointLocation = true,
```

Comment:
Firstly, if **users want** to use temporary checkpoint location for '**other sinks**', it is ok even if we will lost the processed data history. User has the control of  sql behavior.

Secondly, If users want to **continue** the process of data even  for` ConsoleSink` under the condition of 'set checkpoint location', we should give the freedom to users. Again, users has the control of sql behavior.